### ISSUE TITLE
Question - searchbar refs

### ISSUE THREAD

**Issue by nonotest** [react-native-training/react-native-elements#341]
Why are we using a string proptype instead of a function?

**Comment by tarang9211** [comment]
Hey @nonotest. Here are the propTypes the search bar component is using. Which one are you explicity referring to?

```javascript

Search.propTypes = {
  icon: PropTypes.object,
  noIcon: PropTypes.bool,
  lightTheme: PropTypes.bool,
  containerStyle: View.propTypes.style,
  inputStyle: NativeText.propTypes.style,
  round: PropTypes.bool,
  showLoadingIcon: PropTypes.bool,
  loadingIcon: PropTypes.object,
  clearIcon: PropTypes.oneOfType([PropTypes.object, PropTypes.bool]),
  textInputRef: PropTypes.string,
  containerRef: PropTypes.string,
  selectionColor: PropTypes.string,
  underlineColorAndroid: PropTypes.string,
};
```

**Comment by fkrauthan** [comment]
@tarang9211 I believe he is referring to all ref prop types you are defining (same for FormInput ref prop types e.g. textInputRef or containerRef). Using strings for ref is deprecated and the recommended way is using functions (https://zhenyong.github.io/react/docs/more-about-refs.html).

Currently my app is throwing warnings because I follow the react guidelines. So it would be awesome if you could change the prop validation rules to allow functions or strings.

**Comment by nonotest** [comment]
yes sorry.
Thank you Florian.

**Comment by binoy14** [comment]
Should add it for fixing in the next release. Thanks for repoting @nonotest and @fkrauthan for helping out.

**Comment by tarang9211** [comment]
@binoy14 I'll have it fixed, will take the lead on this.

**Comment by tarang9211** [comment]
Guys, the PR for this is https://github.com/react-native-training/react-native-elements/pull/306
Feel free to chime in, let me know what else has to be changed.

**Comment by nonotest** [comment]
Hello,
You might also wanna update the ones from the form input component as
mentioned above.

I don't know if there are any others.

**Comment by tarang9211** [comment]
@nonotest Yea, that's the one I just updated.

**Comment by Monte9** [comment]
I think this was never fixed. Correct me if I am wrong @binoy14.

**Comment by binoy14** [comment]
It is related to the discussion in #362 We need to figure out a proper solution to this

**Comment by Monte9** [comment]
The fix has been merged into `v0.16.0`.
